### ISSUE TITLE
Navigating PhET Menu - No Focus Highlight with Arrow keys 

### ISSUE THREAD

**Issue by terracoda** [phetsims/john-travoltage#212]
@jessegreenberg here are comments about the PhET Menu Button

- I would expect the Arrow to move focus to next menu item, and to cycle through the menu items and circle back to top when the last item is passed.
- VO does identify the pop-up as a menu. W3C has a new example here:
https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-1/menu-button-1.html
- I just tested the above example with and without VO, and it works to spec without VO, but when I turn on VO, I get much less than perfect behavior (also regarding focus), so it could be a VO bug.
- I just checked the PhET Menu again with VO on and I can navigate the items with the Arrow keys, but the focus highlight just does not follow with the Arrow keys. I got the item count and could activate the buttons and links as expected.
- When I get to last menu item, I am told it is the end.
- Using the arrow keys I can sometimes exit the menu or activate the first item (PhET website link). That seems a bit erratic.
- The example menu code you used intentionally does not follow the ARIA spec design pattern. I’m not sure which solution works better for users. If the focus highlights are working for JAWS and NVDA when using the ARROW keys, it is likely a VO bug.

**Comment by terracoda** [comment]
@jessegreenberg, also when I am not using Voice Over:
Visual keyboard focus highlight does not move when I used the Arrow keys to navigate the PhET Menu items (note not using screen reader). When I hit Tab the keyboard focus highlight re-appears, but has obviously skipped items, so virtual focus is moving with Arrow keys, but keyboard focus highlight is not. 

Is the intent that Tab key and the Arrow key when navigating the menu should look the same?

**Comment by jessegreenberg** [comment]
Thanks @terracoda I will respond to each bullet point below:

> I would expect the Arrow to move focus to next menu item, and to cycle through the menu items and circle back to top when the last item is passed.

Yes, it should. Both Tab and arrow keys should go through the items.

> VO does identify the pop-up as a menu.

Ok, great!

> I just tested the above example with and without VO, and it works to spec without VO, but when I turn on VO, I get much less than perfect behavior (also regarding focus), so it could be a VO bug.
I just checked the PhET Menu again with VO on and I can navigate the items with the Arrow keys, but the focus highlight just does not follow with the Arrow keys. I got the item count and could activate the buttons and links as expected.

Hmm, when I use the arrow keys with VoiceOver on, i see the focus highlight flickering.  Maybe focus is fighting with the virtual cursor. I will see if we can fix that.

**Comment by jessegreenberg** [comment]
Luckily, event.preventDefault seems to limit movement of the virtual cursor when using the ```keydown``` event.  After the above commit, behavior seems a bit more consistent when navigating the PhET menu with VoiceOver.

**Comment by jessegreenberg** [comment]
@terracoda the focus is consistent in http://www.colorado.edu/physics/phet/dev/html/john-travoltage/1.3.0-dev.29/john-travoltage_en.html in Safari with VO now when i test, but can you please verify?

**Comment by terracoda** [comment]
@jessegreenberg, this is working much better.

However, pressing the Right and Left Arrow keys should "do nothing." This seems to be true for the Left Arrow, but when I press the Right Arrow something seems to happen...

- With VO off, I lose visual focus when pressing the Right Arrow key. The menu stays open, however. I have to tab away and re-open the window to find visual focus again.  
- With VO on, I also lose focus when pressing the Right Arrow. Pressing the Left Arrow, reads out a number or the font family, but I don't lose focus, and can recover by pressing the Up or Down Arrows. I can't recover after losing focus with the Right Arrow.

Also, any chance the keyboard focus highlight can also include the mouse focus style? Not a show stopper, that might be a longer term general issue we need to work on.

**Comment by jessegreenberg** [comment]
Thanks @terracoda I hadn't noticed that I will take a look at the left/right arrow key issues.

For the mouse hover style, I will take a look and see how difficult it would be to include with focus.

**Comment by jessegreenberg** [comment]
This is now a duplicate of #213, moving recommendation of focus highlight to #213 and closing this issue.
